src/flamegraph/mod.rs

```diff
@@ -424,22 +460,26 @@ fn filled_rectangle<W: Write>(
     y1: usize,
     y2: usize,
     color: (u8, u8, u8),
+    svg_event_empty: &mut Event,
 ) -> quick_xml::Result<usize> {
     let x = write!(buffer, "{}", x1);
     let y = write!(buffer, "{}", y1);
     let width = write!(buffer, "{}", x2 - x1);
     let height = write!(buffer, "{}", y2 - y1);
     let color = write!(buffer, "rgb({},{},{})", color.0, color.1, color.2);
 
-    svg.write_event(Event::Empty(
-        BytesStart::borrowed_name(b"rect").with_attributes(args!(
-                "x" => &buffer[x],
-                "y" => &buffer[y],
-                "width" => &buffer[width],
-                "height" => &buffer[height],
-                "fill" => &buffer[color]
-        )),
-    ))
+    if let Event::Empty(bytes_start) = svg_event_empty {
+        // clear the state
+        bytes_start.clear_attributes();
+        bytes_start.extend_attributes(args!(
+            "x" => &buffer[x],
+            "y" => &buffer[y],
+            "width" => &buffer[width],
+            "height" => &buffer[height],
+            "fill" => &buffer[color]
+        ));
+    }
+    svg.write_event(&svg_event_empty)
 }
```

Comment:
We may want
```rust
} else {
    unreachable!("cache wrapper was of wrong type: {:?}", svg_event_empty);
}
```
And maybe also in the other places where we use this pattern?

Reply:
I thought about

```rust
        if href_is_some {
          svg.write_event(Event::End(BytesEnd::borrowed(b"a")))?;
        }
        svg.write_event(Event::End(BytesEnd::borrowed(b"g")))?;
```
But I am guessing the compiler is good enough to "constantify" those.